src/leet_code/zigzag_conversion.py

```diff
@@ -0,0 +1,19 @@
+class Solution:
+
+    def _zig(self, i, num_rows):
```

Comment:
why is there not a def __init__() ???

Reply:
`class Solution` is the boilerplate class that Leetcode uses in their Python solutions.  Kinda like Code Wars uses a specific function name (and often poorly named) as the entry point to the solution.

That's the what is Solution.  A class I didn't choose, but doesn't answer why no `__init__` method.  A few questions for you, What is the purpose of the `__init__` method, and second, What makes you think there isn't an `__init__` method.

In IPython create a `Solution` class with nothing in it. Then instantiate it and run `dir` on it.  Like so:
```python
class Solution:
    pass

s = Solution()
dir(s)
```
These are the attributes and properties that it has.  Where do they come from?

Reply:
well i guess those attributes and properties come from the __init__ which then would mean that there is an __init__.

Reply:
They don't come from the init.
Consider:
```python
class A:
    def __init__(self, x, y, z):
        self.cubed = x * y * z

    def get_cubed(self):
        return self.cubed

class B(A):
    pass

b = B(1, 2, 3)
dir(b)
# Why does instance `b` have a `get_cubed` method?
print(b.get_cubed())
```

The reason is it inherits from A all of its methods. 

All classes inherit from `object`. This would look like this if written out long hand:
```python
class C(object):
    pass
```

You can run `dir(object)` to see what all is included on the base class.  Which will show an `__init__` method.

Anytime you define a method that is also on the base class, such as an `__init__` method, you are saying don't do what is defined in the base object but rather do what I'm defining here.  The same is True for any method there, like `__repr__`